conans/model/conan_file.py

```diff
@@ -256,20 +257,25 @@ def package_info(self):
         """
 
     def run(self, command, output=True, cwd=None, win_bash=False, subsystem=None, msys_mingw=True,
-            ignore_errors=False, run_environment=False):
-        def _run():
+            ignore_errors=False, run_environment=False, env=None):
+        env = env or {}
+
+        def _run(with_env):
             if not win_bash:
-                return self._runner(command, output, os.path.abspath(RUN_LOG_NAME), cwd)
+                with tools.environment_append(with_env):
+                    return self._runner(command, output, os.path.abspath(RUN_LOG_NAME), cwd)
             # FIXME: run in windows bash is not using output
             return tools.run_in_windows_bash(self, bashcmd=command, cwd=cwd, subsystem=subsystem,
-                                             msys_mingw=msys_mingw)
+                                             msys_mingw=msys_mingw, env=with_env)
         if run_environment:
-            with tools.environment_append(RunEnvironment(self).vars):
-                if os_info.is_macos:
-                    command = 'DYLD_LIBRARY_PATH="%s" %s' % (os.environ.get('DYLD_LIBRARY_PATH', ''), command)
-                retcode = _run()
+            env.update(RunEnvironment(self).vars)
+            if os_info.is_macos:
+                if env.get('DYLD_LIBRARY_PATH'):
```

Comment:
2 gets over DYLID_LIBRARY_PATH, and then using only ``[0]`` item?

Reply:
You are right, I'm not sure what I was thinking.